bokeh/palettes.py

```diff
@@ -1605,13 +1611,17 @@ def varying_alpha_palette(color: str, n: int | None = None, start_alpha: int = 0
     if not (0 <= end_alpha <= 255):
         raise ValueError(f"end_alpha {end_alpha} must be in the range 0 to 255")
 
+    rgba = NamedColor.from_string(color)
+
+    if rgba.a < 1.0:
+        start_alpha = round(start_alpha*rgba.a)
+        end_alpha = round(end_alpha*rgba.a)
+
     if not n:
         n = int(abs(end_alpha - start_alpha)) + 1
 
-    rgb = NamedColor.from_string(color)
-
     diff_alpha = end_alpha - start_alpha
-    palette = tuple(f"{rgb.to_hex()}{round(start_alpha + diff_alpha*i / (n-1.0)):02X}" for i in range(n))
+    palette = tuple(f"{rgba.to_hex()}{round(start_alpha + diff_alpha*i / (n-1.0)):02X}" for i in range(n))
```

Comment:
I think changing the `to_hex()` is a good idea, and it will be easy to see if it breaks any other tests.

There is already a `Color.copy()` function that is overridden in derived classes, but I guess that adding an optional `a` argument wouldn't be ideal. It doesn't really need to use a copy/clone anyway, I can just reuse the one and only `RGB` object and set the alpha component directly.

Reply:
I've have implemented this now.